pkg/reconciler/addressableservice/controller.go

```diff
@@ -43,29 +32,21 @@ func NewController(
 ) *controller.Impl {
 	logger := logging.FromContext(ctx)
 
-	asInformer := asinformer.Get(ctx)
 	svcInformer := svcinformer.Get(ctx)
 
-	c := &Reconciler{
-		Client:        asclient.Get(ctx),
-		Lister:        asInformer.Lister(),
+	r := &Reconciler{
 		ServiceLister: svcInformer.Lister(),
-		Recorder: record.NewBroadcaster().NewRecorder(
-			scheme.Scheme, corev1.EventSource{Component: controllerAgentName}),
 	}
-	impl := controller.NewImpl(c, logger, "AddressableServices")
+	impl := NewImpl(ctx, r)
 
 	logger.Info("Setting up event handlers")
 
-	asInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
```

Comment:
Setting this up automatically is a bad assumption, we used class-filtered enqueuing for Ingress and PodAutoscaler :(